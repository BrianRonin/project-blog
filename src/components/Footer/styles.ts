import styled, { css } from 'styled-components'
import { Main as Text } from '../Text/styles'

export const Main = styled.div`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    background-color: ${theme.colors.white};
    a {
      color: ${theme.colors.darkText};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`
