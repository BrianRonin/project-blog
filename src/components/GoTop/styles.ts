import styled, { css, DefaultTheme } from 'styled-components'

const Light = (theme: DefaultTheme) => css`
  color: ${theme.colors.primary};
  background: ${theme.colors.white};
`
type background = {
  background: boolean
}

export const Main = styled.a<background>`
  ${({ theme, background }) => css`
    position: fixed;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.8;
    border-radius: 50%;
    ${background && Light(theme)};
  `}
`
