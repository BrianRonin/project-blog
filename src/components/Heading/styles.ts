import styled, { css, DefaultTheme } from 'styled-components'
import { HeadingProps } from '.'

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xlarge};
    @media ${theme.media.medium} {
      font-size: ${theme.sizes.large};
    }
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.sizes.xhuge};
    ${mediaFont(theme)}
  `,
}

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.sizes.xlarge};
  }
`

export const Title = styled.h1<
  Pick<HeadingProps, 'light' | 'size' | 'uppercase'>
>`
  ${({ theme, light, size, uppercase }) => css`
    color: ${light ? theme.colors.white : theme.colors.primary};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    ${!!size && titleSize[size](theme)};
  `}
`
