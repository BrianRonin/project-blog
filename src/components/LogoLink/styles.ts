import styled from 'styled-components'
export const Main = styled.a`
  text-decoration: none;
  color: inherit;
  display: flex;
  align-items: center;

  > img {
    width: 7rem;

    border-radius: 50%;
  }
`

export const Logo = styled.img`
  max-height: 6rem;
`
