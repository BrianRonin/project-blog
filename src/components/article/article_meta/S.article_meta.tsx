import styled, { css, DefaultTheme } from 'styled-components'

const name = (theme: DefaultTheme) => css`
  //
`

export const Main = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.medium};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .tags span::after {
      content: ', ';
    }

    .tags span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300msa ease-in-out;
      &:hover {
        filter: brightness(50%);
      }
    }

    @media ${theme.media.medium} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`

export const Span = styled.span`
  ${({ theme }) => css`
    //
  `}
`
