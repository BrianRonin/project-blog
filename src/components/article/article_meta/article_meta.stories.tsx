import {
  Meta,
  Story,
} from '@storybook/react/types-6-0'
import {
  ArticleMeta,
  articleMetaProps,
} from './article_meta'
import { mock_article_meta } from './M.article_meta'

export default {
  title: 'components/article/Article meta',
  component: ArticleMeta,
  args: mock_article_meta,
  argTypes: {
    createAt: {
      control: {
        type: 'date',
      },
    },
  },
} as Meta

export const Template: Story<articleMetaProps> = (
  args,
) => <ArticleMeta {...args} />
export const noAuthorAndTags: Story<
  articleMetaProps
> = (args) => (
  <ArticleMeta
    {...args}
    tags={undefined}
    author={undefined}
  />
)

Template.parameters = {
  backgrounds: {
    default: 'light',
  },
}
noAuthorAndTags.parameters = {
  backgrounds: {
    default: 'light',
  },
}
