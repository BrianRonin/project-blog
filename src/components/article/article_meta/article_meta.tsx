import { Text } from '../../Text'
import * as S from './S.article_meta'
import { formatDate } from '../../../utils/format-date'
import { type_strapi_author } from '../../../types/strapi/author'
import { type_strapi_category } from '../../../types/strapi/category'
import { type_strapi_tags } from '../../../types/strapi/tags'

export type articleMetaProps = {
  createdAt: string
  author: type_strapi_author
  tags: type_strapi_tags[]
  category: type_strapi_category
}
export const ArticleMeta = ({
  createdAt,
  author,
  tags,
  category,
}: articleMetaProps) => {
  return (
    <S.Main className='id-component-article-meta'>
      <S.Span>
        <p>
          Por{' '}
          <a href={'/author/' + author.slug}>
            <strong>{author.name}</strong>
          </a>{' '}
          em <time dateTime={createdAt}>{formatDate(createdAt)}, </time>
          {!!category && (
            <a href={`/category/${category.slug}`}>
              <strong>{category.name + ', '}</strong>
            </a>
          )}
          <span className='tags'>
            {!!tags &&
              tags.map((tag, index) => (
                <span key={`tag-blog-id-${index}`}>
                  <a href={`/tags/${tag.slug}`}>{tag.name}</a>
                </span>
              ))}
          </span>
        </p>
      </S.Span>
    </S.Main>
  )
}
