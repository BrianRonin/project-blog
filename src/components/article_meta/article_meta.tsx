import { Text } from '../Text'
import * as S from './S.article_meta'
import { formatDate } from '../../utils/format-date'
// data autor tags
export type articleMetaProps = {
  createdAt: string
  author: { slug: string; name: string }
  tags: { name: string; slug: string }[]
  category: { name: string; slug: string }
}
export const ArticleMeta = ({
  createdAt,
  author,
  tags,
  category,
}: articleMetaProps) => {
  return (
    <S.Main className='id-component-article-meta'>
      <S.Span>
        <p>
          Por{' '}
          <a href={'/author/' + author.slug}>
            <strong>{author.name}</strong>
          </a>{' '}
          em <time dateTime={createdAt}>{formatDate(createdAt)}, </time>
          {!!category && (
            <a href={`/category/${category.slug}`}>
              <strong>{category.name + ', '}</strong>
            </a>
          )}
          <span className='tags'>
            {!!tags &&
              tags.map((tag, index) => (
                <span key={`tag-blog-id-${index}`}>
                  <a href={`/tags/${tag.slug}`}>{tag.name}</a>
                </span>
              ))}
          </span>
        </p>
      </S.Span>
    </S.Main>
  )
}
