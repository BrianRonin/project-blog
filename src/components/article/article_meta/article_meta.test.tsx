import { screen } from '@testing-library/react'
import { renderTheme } from '~/styles/render-theme'
import { ArticleMeta } from './article_meta'
import { mock_article_meta as mock } from './M.article_meta'

describe('<ArticleMeta />', () => {
  it('should render name author and link', () => {
    const { debug, container } = renderTheme(<ArticleMeta {...mock} />)
    debug()
    expect(
      screen.getByRole('link', { name: mock.author.name }),
    ).toHaveAttribute('href', '/author/' + mock.author.slug)
    const date = screen.getByText(/4 de out. de 2022/)
    console.log(date)
    expect(date).toHaveAttribute('datetime', mock.createdAt)
    expect(
      screen.getByRole('link', { name: RegExp(mock.category.name) }),
    ).toHaveAttribute('href', '/category/' + mock.category.slug)
    const tags = document.getElementsByClassName('tags')
    expect(tags.length).toBe(1)
    expect(tags[0].textContent).toMatch(/#React#JavaScript/)
    expect(container).toMatchSnapshot()
  })
})
