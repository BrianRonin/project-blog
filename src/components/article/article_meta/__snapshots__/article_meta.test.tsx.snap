// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta /> should render name author and link 1`] = `
.c0 {
  font-size: 2.4rem;
  color: #4f4f4fff;
  font-style: italic;
}

.c0 .tags span::after {
  content: ', ';
}

.c0 .tags span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300msa ease-in-out;
  transition: all 300msa ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

@media (max-width:768px) {
  .c0 {
    font-size: 1.6rem;
  }
}

<div>
  <div
    class="c0 id-component-article-meta"
  >
    <span
      class=""
    >
      <p>
        Por
         
        <a
          href="/author/brian"
        >
          <strong>
            BrianRonin
          </strong>
        </a>
         
        em 
        <time
          datetime="2022-10-05T02:48:15.628Z"
        >
          4 de out. de 2022
          , 
        </time>
        <a
          href="/category/docker"
        >
          <strong>
            Docker, 
          </strong>
        </a>
        <span
          class="tags"
        >
          <span>
            <a
              href="/tags/react"
            >
              #React
            </a>
          </span>
          <span>
            <a
              href="/tags/java-script"
            >
              #JavaScript
            </a>
          </span>
        </span>
      </p>
    </span>
  </div>
</div>
`;
